### PR TITLE
feat(cart): Add Composite Product to Cart

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

## Other information
It is a breaking change, because before the code assumed that all product added to the cart followed the same workflow. That is no longer the case and types need to be added to every add to cart call. Composite products also have their own unique required fields.

### CODE REVIEW

File: libs/cart/src/drivers/magento/cart-item.service.ts [hunk 1]
```diff
@@ -55,16 +56,12 @@ export class DaffMagentoCartItemService implements DaffCartItemServiceInterface
...
-      }
-    }).pipe(
-      map(result => this.cartTransformer.transform(result.data.addSimpleProductsToCart.cart))
-    )
+  add(cartId: string, cartItemInput: DaffCartItemInput): Observable<Partial<DaffCart>> {
+		if(cartItemInput.type === DaffCartItemInputType.Composite) {
+			return this.addBundledProduct(cartId, <DaffCompositeCartItemInput>cartItemInput);
+		} else {
+			return this.addSimpleProduct(cartId, cartItemInput);
+		}
```
Comment (griest024) [inline libs/cart/src/drivers/magento/cart-item.service.ts:64]: I think these conditionals are better expressed as a switch statement.

Comment (lderrickable) [reply libs/cart/src/drivers/magento/cart-item.service.ts]: changed

File: libs/cart/src/drivers/magento/cart-item.service.ts [hunk 2]
```diff
@@ -43,9 +43,10 @@ export class DaffMagentoCartItemService implements DaffCartItemServiceInterface
       query: listCartItems,
       variables: {cartId}
     }).pipe(
-      map(result =>
-        result.data.cart.items.map(item => this.cartItemTransformer.transform(item))
-      )
+      map(result => {
+				console.log(result);
+        return result.data.cart.items.map(item => this.cartItemTransformer.transform(item))
+			})
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/cart/src/drivers/magento/cart-item.service.ts]: done

File: libs/cart/src/drivers/magento/transforms/inputs/cart-item-input-transformers.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { MagentoCartItemInput, MagentoBundledCartItemInput, MagentoBundledCartItemOption } from '../../models/inputs/cart-item';
+import { DaffCartItemInput, DaffCompositeCartItemInput, DaffCompositeCartItemOption } from '../../../../models/cart-item-input';
+
+export function transformCompositeCartItem(item: DaffCompositeCartItemInput): MagentoBundledCartItemInput {
+	return {
+		input: transformSimpleCartItem(item),
+		options: item.options ? item.options.map(transformCompositeCartItemOption) : null
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/cart/src/drivers/magento/transforms/inputs/cart-item-input-transformers.ts]: changed

File: libs/cart/src/drivers/magento/transforms/inputs/cart-item-input-transformers.spec.ts [hunk 4]
```diff
@@ -52,16 +52,20 @@ describe('Driver | Magento | Cart | Transformers | MagentoCartItemInput', () =>
...
       expect(transformedCartItem.options[0].id).toEqual(Number(mockDaffCompositeCartItemInput.options[0].id));
       expect(transformedCartItem.options[0].quantity).toEqual(mockDaffCompositeCartItemInput.options[0].quantity);
       expect(transformedCartItem.options[0].value).toEqual([mockDaffCompositeCartItemInput.options[0].value]);
-    });
+		});
+		
+		it('should return the correct values when options is null', () => {
+			mockDaffCompositeCartItemInput.options = null;
+			transformedCartItem = transformCompositeCartItem(mockDaffCompositeCartItemInput);
+		});
```
Comment (griest024) [inline libs/cart/src/drivers/magento/transforms/inputs/cart-item-input-transformers.spec.ts:69]: I think you forgot to make an expectation here.

Comment (lderrickable) [reply libs/cart/src/drivers/magento/transforms/inputs/cart-item-input-transformers.spec.ts]: oops. Added

**Pull request merged**